### PR TITLE
story(ccls-1956) get/put application correspondence address

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**

### CODE REVIEW

File: caab-api/open-api-specification.yml [hunk 1]
```diff
@@ -156,13 +156,13 @@ paths:
...
+    put:
       tags:
         - applications
-      summary: 'patch an applications provider details'
-      operationId: 'patchApplicationProviderDetails'
+      summary: 'put an applications provider details'
+      operationId: 'putApplicationProviderDetails'
       requestBody:
-        description: patch an application's application type
+        description: put an application's application type
```
Comment (porritta) [inline caab-api/open-api-specification.yml:165]: Update description for provider details

File: caab-api/open-api-specification.yml [hunk 2]
```diff
@@ -175,7 +175,72 @@ paths:
             type: 'integer'
             format: 'int64'
             example: '1234567890'
-        - name: 'Caab-user-Login-Id'
+        - name: 'Caab-User-Login-Id'
+          in: header
+          required: true
+          schema:
+            type: 'string'
+          example: 'MARILYN@DESORANDCO.CO.UK'
```
Comment (porritta) [inline caab-api/open-api-specification.yml:183]: Isn't this a real provider login? Maybe shouldn't be used as an example.

File: caab-api/open-api-specification.yml (hunk 1 above)
Comment (PhilDigitalJustice) [reply caab-api/open-api-specification.yml]: fixed

File: caab-api/open-api-specification.yml (hunk 2 above)
Comment (PhilDigitalJustice) [reply caab-api/open-api-specification.yml]: amended

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java [hunk 3]
```diff
@@ -92,20 +105,38 @@ public ApplicationProviderDetails getApplicationProviderDetails(final Long id) {
...
   /**
-   * Patch an application with new provider details.
+   * Adds/Updates an application with a new correspondence address.
+   *
+   * @param id the TDS id for the application.
+   * @param address the applications updated correspondence address
+   */
+  @Transactional
+  public void putCorrespondenceAddress(
+      final Long id,
```
Comment (porritta) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:122]: Could we rename this attribute `applicationId` to make it clearer?

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java [hunk 4]
```diff
@@ -92,20 +105,38 @@ public ApplicationProviderDetails getApplicationProviderDetails(final Long id) {
...
+  }
+
+  /**
+   * Amends an application with new provider details.
    *
    * @param id the TDS id for the application.
    * @param providerDetails the new application provider details to update the application with
    */
-  public void patchProviderDetails(
+  public void putProviderDetails(
```
Comment (porritta) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:139]: Same comment here, re: applicationId

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java [hunk 5]
```diff
@@ -118,12 +149,12 @@ public void patchProviderDetails(
...
 
   /**
-   * Patch an application with new application type.
+   * Amends an application with new application type.
    *
    * @param id the TDS id for the application.
    * @param applicationType the new application type to update the application with
    */
-  public void patchApplicationType(
+  public void putApplicationType(
```
Comment (porritta) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:157]: Same comment here, re: applicationId

Comment (PhilDigitalJustice) [reply caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:164]: amended

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java (hunk 4 above)
Comment (PhilDigitalJustice) [reply caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:145]: amended

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java (hunk 3 above)
Comment (PhilDigitalJustice) [reply caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java]: amended

**Review by porritta — APPROVED** [review]
LGTM

**Pull request merged**